### DISCUSSION TITLE
I want to contribute, can I change index.js and index.es.js directly or it's auto generated

### DISCUSSION THREAD

**Discussion by rluvaton** [FortAwesome/Font-Awesome#18881] [Q&A]
I want to contribute, can I change index.js and index.es.js in the js packages directly or it's auto-generated?

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

It is not possible to directly change the source files because they are generated from an upstream repository

However, if you contribute by fixing an issue, we may take into account the change to provide a faster solution

**Reply by rluvaton** [reply]
Not an issue but an improvement, currently in the index files the entire icons are there, dead code elimination remove what you don't use but the source map containing all the icons definitions that in the index file, I want to change that instead of declaring every icon in the index file we just export the icons in the icons file, this way tree shaking will remove what not being used and the source map will be reduced and everyone will be happy 😊

**Reply by tagliala** [reply]
Hi, apologies for the late reply

As a non-javascript developer I can't fully understand this change

However, I see in my js sources a list of all the Font Awesome icons, even those that I'm not using (and not importing)

```js
/* harmony export */   "faBuildingShield": function() { return /* binding */ faBuildingShield; },
/* harmony export */   "faBuildingUn": function() { return /* binding */ faBuildingUn; },
/* harmony export */   "faBuildingUser": function() { return /* binding */ faBuildingUser; },
/* harmony export */   "faBuildingWheat": function() { return /* binding */ faBuildingWheat; },
/* harmony export */   "faBullhorn": function() { return /* binding */ faBullhorn; },
/* harmony export */   "faBullseye": function() { return /* binding */ faBullseye; },
/* harmony export */   "faBurger": function() { return /* binding */ faBurger; },
/* harmony export */   "faBurn": function() { return /* binding */ faBurn; },
/* harmony export */   "faBurst": function() { return /* binding */ faBurst; },
/* harmony export */   "faBus": function() { return /* binding */ faBus; },
/* harmony export */   "faBusAlt": function() { return /* binding */ faBusAlt; },
```

So I'm asking if the proposed change has something to do with this issue
